project/src/main/kotlin/org/ionproject/core/writeApi/insertClassSectionEvents/InsertClassSectionEventsController.kt

```diff
@@ -0,0 +1,106 @@
+package org.ionproject.core.writeApi.insertClassSectionEvents
+
+import com.fasterxml.jackson.databind.JsonNode
+import org.ionproject.core.common.Media
+import org.ionproject.core.common.ProblemJson
+import org.ionproject.core.join
+import org.ionproject.core.writeApi.common.Uri
+import org.ionproject.core.writeApi.insertClassSectionEvents.json.SchemaValidator
+import org.ionproject.core.writeApi.insertClassSectionEvents.json.InsertClassSectionEventsParams
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException
+import org.postgresql.util.PSQLException
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class InsertClassSectionEventsController(private val repo: InsertClassSectionEventsRepo) {
+
+  @PutMapping(Uri.insertClassSectionEvents, consumes = [Media.APPLICATION_JSON])
+  fun insertClassSectionEvents(@RequestBody json: JsonNode): ResponseEntity<Any> {
+
+    /**
+     * Analyze the received JSON object in comparison with the defined
+     * JSON Schema (statically defined JSON object which enforces syntax and semantic restrictions on the requests).
+     */
+    val errMessages = SchemaValidator.validate(json)
+    if (errMessages.isNotEmpty()) {
+      // Invalid JSON received
+      return ResponseEntity.badRequest().body(
```

Comment:
Wouldn't it be better to avoid repeating code adding this as an exception to the CustomExceptions folder?

Reply:
I'm not sure what you mean. No exception is being thrown here.